### ISSUE TITLE
Re-Balancing Connections when using a hash-based load balancer

### ISSUE THREAD

**Issue by asgoel** [envoyproxy/envoy#2819]
Been exploring the envoy docs a little bit and haven't seen anything on this topic so thought I would ask. If I have a load balancer set up that uses the ring-hash algorithm, how does re-balancing work, with regards to upstreams coming up or down?

Specifically, if I have upstream A and upstream B, with persistent connections (say, websockets) open to both, and A goes down, will all traffic be routed to B? What happens when A comes back up? Will the LB kill any connections that are open to B that are meant to go to A? Or will it keep those connections open and only **new** connections will go to A?

Basically, we want to guarantee that for a specific header value, **all** clients are **always** connected to the same upstream, regardless of what the upstream actually is.

**Comment by mattklein123** [comment]
It's the former. No existing requests are effected during rehashing. For long lives request reams (websocket/gRPC) you have a split brain situation. It's possible we could kill all connections on rehash, but that would be a feature request and need a bunch of thinking.

**Comment by mattklein123** [comment]
Closing as answered. Please feel free to open a feature request issue with more details on the requirements and use cases.
